sphinx/environment/__init__.py

```diff
@@ -668,6 +669,10 @@ def read_doc(self, docname, app=None):
         self.settings['trim_footnote_reference_space'] = \
             self.config.trim_footnote_reference_space
         self.settings['gettext_compact'] = self.config.gettext_compact
+        language = (self.config.language or 'en').replace('_', '-')
+        self.settings['language_code'] = language
+        if language in smartchars.quotes:
+            self.settings['smart_quotes'] = self.config.html_use_smartypants
```

Comment:
I don’t see why one may want this behavior for HTML builder and not for other builders. Actually even the existing code applied to LaTeX as well as HTML (see the `educate_quotes_latex` function which I removed).

However if we want to rename the setting (and thus break compatibility), we may as well extend it with the other options Docutils provides (i.e. add support for `use_smartypants = 'alt'`), or remove it altogether as suggested by @gmilde on [sphinx-dev group][1] (if we remove it we need to make sure smart quotes is still enabled by default).

@tk0miya What do you think? Should we remove this setting?

[1]: https://groups.google.com/d/msg/sphinx-dev/6zwDVG8UG7Q/cpdx_xVdAQAJ